File: frontend_flutter/lib/models/favorite.dart
class Favorite {
  final String id;
  final String messageId;
  final String question;
  final String answer;
  final DateTime createdAt;

  const Favorite({
    required this.id,
    required this.messageId,
    required this.question,
    required this.answer,
    required this.createdAt,
  });

  factory Favorite.fromJson(Map<String, dynamic> json) => Favorite(
        id: json['id']?.toString() ?? '',
        messageId: json['message_id']?.toString() ?? '',
        question: json['question'] ?? '',
        answer: json['answer'] ?? '',
        createdAt: DateTime.tryParse(json['created_at'] ?? '') ?? DateTime.now(),
      );

  Map<String, dynamic> toLocalDb() => {
        'id': id,
        'message_id': messageId,
        'question': question,
        'answer': answer,
        'created_at': createdAt.toIso8601String(),
      };

  factory Favorite.fromLocalDb(Map<String, dynamic> row) => Favorite(
        id: row['id'] as String,
        messageId: row['message_id'] as String,
        question: row['question'] as String,
        answer: row['answer'] as String,
        createdAt: DateTime.parse(row['created_at'] as String),
      );
}

File: backend/app/models/favorite.py
"""
收藏夹数据模型
用户收藏的问答对，方便日后查阅
"""

from sqlalchemy import Column, Integer, Text, DateTime, ForeignKey, func
from app.database import Base


class Favorite(Base):
    """收藏夹表"""
    __tablename__ = "favorites"

    id = Column(Integer, primary_key=True, autoincrement=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False, index=True)
    message_id = Column(Integer, ForeignKey("messages.id"), nullable=False)
    question = Column(Text, nullable=False)   # 冗余存储，方便展示无需 join
    answer = Column(Text, nullable=False)
    sources = Column(Text, nullable=True)     # JSON 字符串
    created_at = Column(DateTime, server_default=func.now())

    def __repr__(self):
        return f"<Favorite(id={self.id}, user_id={self.user_id})>"
